solidity/contracts/middleware/InterchainAccountRouter.sol

```diff
@@ -150,14 +163,16 @@ contract InterchainAccountRouter is Router {
      * @param _to The address of the contract to call
      * @param _value The value to include in the call
      * @param _data The calldata
+     * @param _hookMetadata The hook metadata to override with for the hook set by the owner
      * @return The Hyperlane message ID
      */
     function callRemote(
         uint32 _destination,
         address _to,
         uint256 _value,
-        bytes memory _data
-    ) external returns (bytes32) {
+        bytes memory _data,
+        bytes memory _hookMetadata
+    ) external payable returns (bytes32) {
```

Comment:
can we keep these functions for backwards compatibility and just add new ones with hook metadata?

Reply:
done